analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.Monitor.Loops.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+
+class Program
+{
+    private object obj = new object();
+
+    public void Method1()
+    {
+        Monitor.Enter(obj);
+        for (int i = 0; i < 10; i++)
+        {
+            if (i == 9)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+    }
+
+    public void Method2()
+    {
+        Monitor.Enter(obj); // FN
+        for (int i = 0; i < 10; i++)
+        {
+            break;
+            if (i == 9)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+    }
+
+    public void Method3()
+    {
+        Monitor.Enter(obj); // FN
+        for (int i = 0; i < 10; i++)
+        {
+            if (i == 5)
+            {
+                break;
+            }
+
+            if (i == 9)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+    }
+
+    public void Method4()
+    {
+        Monitor.Enter(obj);
+        for (int i = 0; i < 10; i++)
+        {
+            if (i == 10)
+            {
+                break;
+            }
+
+            if (i == 9)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+    }
+
+    public void Method5()
+    {
+        Monitor.Enter(obj); // FN
+        for (int i = 0; i < 10; i++)
+        {
+            if (i == 9)
+            {
+                continue;
+            }
+
+            if (i == 9)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+    }
+
+    public void Method6(bool condition, byte[] array)
+    {
+        Monitor.Enter(obj); // FN
+        foreach (var item in array)
+        {         
+            if (condition)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+    }
+
+    public void Method7(bool condition, byte[] array)
+    {
+        Monitor.Enter(obj); // FN
```

Comment:
To explain the reasoning. It's unusual
```suggestion
        Monitor.Enter(obj); // FN, array can be empty
```

Reply:
Same for others below

Reply:
Line 131 looks unfinished

Reply:
Good catch on line 145